packages/vite/src/node/http.ts

```diff
@@ -178,8 +178,19 @@ export async function httpServerStart(
     httpServer.on('error', onError)
 
     httpServer.listen(port, host, () => {
-      httpServer.removeListener('error', onError)
-      resolve(port)
+      if (host === 'localhost') {
```

Comment:
I think we should listen to `0.0.0.0` first and then listen to the specified host to avoid other process to listen to that port between switching the host.

Also I guess we need to check `0.0.0.0` for hosts other than localhost too.

Reply:
Hi sapphi-red, I have a fix for your suggestion here: https://github.com/Scttpr/vite/pull/3 I am unsure how I should move forward with this